File: laravel-app/app/Models/Invoice.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;

class Invoice extends Model
{
    use HasFactory;

    protected $fillable = [
        'invoice_number',
        'stock_out_id',
        'customer_id',
        'invoice_date',
        'subtotal',
        'discount',
        'additional_cost',
        'grand_total',
        'paid_amount',
        'due_amount',
        'status_id',
        'created_by',
    ];

    protected function casts(): array
    {
        return [
            'invoice_date' => 'date',
            'subtotal' => 'decimal:2',
            'discount' => 'decimal:2',
            'additional_cost' => 'decimal:2',
            'grand_total' => 'decimal:2',
            'paid_amount' => 'decimal:2',
            'due_amount' => 'decimal:2',
        ];
    }

    public function stockOut(): BelongsTo
    {
        return $this->belongsTo(StockOut::class);
    }

    public function customer(): BelongsTo
    {
        return $this->belongsTo(Customer::class)->withTrashed();
    }

    public function status(): BelongsTo
    {
        return $this->belongsTo(Status::class);
    }

    public function creator(): BelongsTo
    {
        return $this->belongsTo(User::class, 'created_by')->withTrashed();
    }

    public function items(): HasMany
    {
        return $this->hasMany(InvoiceItem::class);
    }

    public function getPaymentStatusAttribute(): string
    {
        if ((float) $this->due_amount <= 0) {
            return 'paid';
        }

        if ((float) $this->paid_amount > 0) {
            return 'partial';
        }

        return 'due';
    }
}

File: mobile-app/lib/models/invoice.dart
import 'customer.dart';
import 'line_item.dart';
import 'lookup.dart';

class InvoiceModel {
  final int id;
  final String invoiceNumber;
  final int? saleId;
  final CustomerRef? customer;
  final String invoiceDate;
  final double subtotal;
  final double discount;
  final double additionalCost;
  final double grandTotal;
  final double paidAmount;
  final double dueAmount;
  final String? paymentStatus;
  final StatusModel? status;
  final String? createdBy;
  final List<LineItem> items;
  final String? createdAt;

  InvoiceModel({
    required this.id,
    required this.invoiceNumber,
    this.saleId,
    this.customer,
    required this.invoiceDate,
    required this.subtotal,
    required this.discount,
    required this.additionalCost,
    required this.grandTotal,
    required this.paidAmount,
    required this.dueAmount,
    this.paymentStatus,
    this.status,
    this.createdBy,
    this.items = const [],
    this.createdAt,
  });

  factory InvoiceModel.fromJson(Map<String, dynamic> json) => InvoiceModel(
        id: json['id'] as int,
        invoiceNumber: json['invoice_number'] as String,
        saleId: (json['sale_id'] as num?)?.toInt(),
        customer: json['customer'] != null ? CustomerRef.fromJson(json['customer'] as Map<String, dynamic>) : null,
        invoiceDate: json['invoice_date'] as String,
        subtotal: (json['subtotal'] as num).toDouble(),
        discount: (json['discount'] as num).toDouble(),
        additionalCost: (json['additional_cost'] as num).toDouble(),
        grandTotal: (json['grand_total'] as num).toDouble(),
        paidAmount: (json['paid_amount'] as num).toDouble(),
        dueAmount: (json['due_amount'] as num).toDouble(),
        paymentStatus: json['payment_status'] as String?,
        status: json['status'] != null ? StatusModel.fromJson(json['status'] as Map<String, dynamic>) : null,
        createdBy: json['created_by'] as String?,
        items: (json['items'] as List<dynamic>? ?? [])
            .map((e) => LineItem.fromJson(e as Map<String, dynamic>))
            .toList(),
        createdAt: json['created_at'] as String?,
      );
}
